diff --git a/include/stable-diffusion.h b/include/stable-diffusion.h
index 51b2b32..8da9adf 100644
--- a/include/stable-diffusion.h
+++ b/include/stable-diffusion.h
@@ -331,10 +331,12 @@ typedef struct sd_ctx_t sd_ctx_t;
 
 typedef void (*sd_log_cb_t)(enum sd_log_level_t level, const char* text, void* data);
 typedef void (*sd_progress_cb_t)(int step, int steps, float time, void* data);
+typedef bool (*sd_abort_cb_t)(void* data);
 typedef void (*sd_preview_cb_t)(int step, int frame_count, sd_image_t* frames, bool is_noisy, void* data);
 
 SD_API void sd_set_log_callback(sd_log_cb_t sd_log_cb, void* data);
 SD_API void sd_set_progress_callback(sd_progress_cb_t cb, void* data);
+SD_API void sd_set_abort_callback(sd_abort_cb_t cb, void* data);
 SD_API void sd_set_preview_callback(sd_preview_cb_t cb, enum preview_t mode, int interval, bool denoised, bool noisy, void* data);
 SD_API int32_t sd_get_num_physical_cores();
 SD_API const char* sd_get_system_info();
diff --git a/src/stable-diffusion.cpp b/src/stable-diffusion.cpp
index d769d45..d51f039 100644
--- a/src/stable-diffusion.cpp
+++ b/src/stable-diffusion.cpp
@@ -2193,6 +2193,9 @@ public:
                 int showstep = std::abs(step);
                 pretty_progress(showstep, (int)steps, (t1 - t0) / 1000000.f / showstep);
                 // LOG_INFO("step %d sampling completed taking %.2fs", step, (t1 - t0) * 1.0f / 1000000);
+                if (sd_abort_requested()) {
+                    return (ggml_tensor*)nullptr;
+                }
             }
             return denoised;
         };
diff --git a/src/util.cpp b/src/util.cpp
index a94cfd9..343815d 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -270,6 +270,9 @@ int32_t sd_get_num_physical_cores() {
 static sd_progress_cb_t sd_progress_cb = nullptr;
 void* sd_progress_cb_data              = nullptr;
 
+static sd_abort_cb_t sd_abort_cb = nullptr;
+static void* sd_abort_cb_data   = nullptr;
+
 static sd_preview_cb_t sd_preview_cb = nullptr;
 static void* sd_preview_cb_data      = nullptr;
 preview_t sd_preview_mode            = PREVIEW_NONE;
@@ -423,6 +426,15 @@ void sd_set_progress_callback(sd_progress_cb_t cb, void* data) {
     sd_progress_cb      = cb;
     sd_progress_cb_data = data;
 }
+
+void sd_set_abort_callback(sd_abort_cb_t cb, void* data) {
+    sd_abort_cb      = cb;
+    sd_abort_cb_data = data;
+}
+
+bool sd_abort_requested() {
+    return sd_abort_cb && sd_abort_cb(sd_abort_cb_data);
+}
 void sd_set_preview_callback(sd_preview_cb_t cb, preview_t mode, int interval, bool denoised, bool noisy, void* data) {
     sd_preview_cb       = cb;
     sd_preview_cb_data  = data;
diff --git a/src/util.h b/src/util.h
index 7dee7bf..254041e 100644
--- a/src/util.h
+++ b/src/util.h
@@ -69,6 +69,7 @@ protected:
 std::string path_join(const std::string& p1, const std::string& p2);
 std::vector<std::string> split_string(const std::string& str, char delimiter);
 void pretty_progress(int step, int steps, float time);
+bool sd_abort_requested();
 
 void log_printf(sd_log_level_t level, const char* file, int line, const char* format, ...);
 
