diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ef3a1308..ecf0f4a1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -249,7 +249,9 @@ function(ggml_add_backend_library backend)
     if (GGML_BACKEND_DL AND NOT (GGML_CPU_STATIC AND ${backend} MATCHES "^ggml-cpu"))
         add_library(${backend} MODULE ${ARGN})
         # write the shared library to the output directory
-        set_target_properties(${backend} PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+        set_target_properties(${backend} PROPERTIES
+            LIBRARY_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
+            OUTPUT_NAME "qvac-diffusion-${backend}")
         target_compile_definitions(${backend} PRIVATE GGML_BACKEND_DL)
         add_dependencies(ggml ${backend})
         if (GGML_BACKEND_DIR)
diff --git a/src/ggml-backend-reg.cpp b/src/ggml-backend-reg.cpp
index 8a693f84..b5405bfd 100644
--- a/src/ggml-backend-reg.cpp
+++ b/src/ggml-backend-reg.cpp
@@ -437,9 +437,9 @@ static fs::path get_executable_path() {
 
 static fs::path backend_filename_prefix() {
 #ifdef _WIN32
-    return fs::u8path("ggml-");
+    return fs::u8path("qvac-diffusion-ggml-");
 #else
-    return fs::u8path("libggml-");
+    return fs::u8path("libqvac-diffusion-ggml-");
 #endif
 }
 
